### ISSUE TITLE
[feature] MSBuild add definitions for ResourceCompile

### ISSUE THREAD

**Issue by taogeng** [conan-io/conan#9828]
MSBuild build method parameter of definitions only works for ClCompile,  and I wish a parameter of definitions, such as  res_definitions for ResourceCompile

**Comment by memsharded** [comment]
Hi @taogeng 

First, you should start using the new integrations from ``from conan.tools.microsoft``, not the legacy ones: https://docs.conan.io/en/latest/reference/conanfile/tools.html

Then, please you need to provide more details. Like a reproducible example, with the obtained output (the .props file, for example), and then the desired, full output that you would like to see (the .props full changes).

**Comment by taogeng** [comment]
I want to specify definetions for both C/C++ -> Preprocessor AND Resources -> General -> Preprocessor Definitaions items.

here is the example repos:
1. git clone https://github.com/taogeng/conan-test.git
2. change directory to /conan-test
2. conan install .
3. conan build .

and there will be a compile error.

**Comment by memsharded** [comment]
Hi @taogeng 

You are not using the modern integrations, you are still using the legacy. You should do something more like:

```python
from conans import ConanFile
from conan.tools.microsoft import MSBuild, MSBuildToolchain


class TestConan(ConanFile):
    name = "test"
    version = "1.0.0"
    settings = "os", "compiler", "build_type", "arch"

    def generate(self):
        tc = MSBuildToolchain(self)
        tc.preprocessor_definitions["ROLE_APP1"] = "Something"
        tc.generate()

    def build(self):
        msbuild = MSBuild(self)
        msbuild.build("test.sln", upgrade_project=False, verbosity="normal")
```

Note the import from ``from conan.tools.microsoft``.

This will generate a ``conantoolchain.props`` files that you need to add to your .vcxproj in order to be able to inject arbitrary preprocessor definition.

Once you have this, please report on the current output of ``conantoolchain_release_x64.props`` (currently something like):
```
<?xml version="1.0" encoding="utf-8"?>
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <ItemDefinitionGroup>
    <ClCompile>
      <PreprocessorDefinitions>
         ROLE_APP1=Something;%(PreprocessorDefinitions)
      </PreprocessorDefinitions>
      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
      <LanguageStandard></LanguageStandard>
    </ClCompile>
  </ItemDefinitionGroup>
  <PropertyGroup Label="Configuration">
    <PlatformToolset>v142</PlatformToolset>
  </PropertyGroup>
</Project>
```
And then suggest what should be different and why.

**Comment by taogeng** [comment]
> Hi @taogeng
> 
> You are not using the modern integrations, you are still using the legacy. You should do something more like:
> 
> ```python
> from conans import ConanFile
> from conan.tools.microsoft import MSBuild, MSBuildToolchain
> 
> 
> class TestConan(ConanFile):
>     name = "test"
>     version = "1.0.0"
>     settings = "os", "compiler", "build_type", "arch"
> 
>     def generate(self):
>         tc = MSBuildToolchain(self)
>         tc.preprocessor_definitions["ROLE_APP1"] = "Something"
>         tc.generate()
> 
>     def build(self):
>         msbuild = MSBuild(self)
>         msbuild.build("test.sln", upgrade_project=False, verbosity="normal")
> ```
> 
> Note the import from `from conan.tools.microsoft`.
> 
> This will generate a `conantoolchain.props` files that you need to add to your .vcxproj in order to be able to inject arbitrary preprocessor definition.
> 
> Once you have this, please report on the current output of `conantoolchain_release_x64.props` (currently something like):
> 
> ```
> <?xml version="1.0" encoding="utf-8"?>
> <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
>   <ItemDefinitionGroup>
>     <ClCompile>
>       <PreprocessorDefinitions>
>          ROLE_APP1=Something;%(PreprocessorDefinitions)
>       </PreprocessorDefinitions>
>       <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
>       <LanguageStandard></LanguageStandard>
>     </ClCompile>
>   </ItemDefinitionGroup>
>   <PropertyGroup Label="Configuration">
>     <PlatformToolset>v142</PlatformToolset>
>   </PropertyGroup>
> </Project>
> ```
> 
> And then suggest what should be different and why.

It still doesn't solve my problem. The `conantoolchain_release_win32.props` file should be like as follows with ResourceCompile items:
```
<?xml version="1.0" encoding="utf-8"?>
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <ItemDefinitionGroup>
    <ClCompile>
      <PreprocessorDefinitions>
         ROLE_APP1=Something;%(PreprocessorDefinitions)
      </PreprocessorDefinitions>
      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
      <LanguageStandard></LanguageStandard>
    </ClCompile>
	<ResourceCompile>
	  <PreprocessorDefinitions>
		 ROLE_APP1=Something;%(PreprocessorDefinitions)
	  </PreprocessorDefinitions>
	</ResourceCompile>
  </ItemDefinitionGroup>
  <PropertyGroup Label="Configuration">
    <PlatformToolset>v142</PlatformToolset>
  </PropertyGroup>
</Project>

```

**Comment by memsharded** [comment]
Now I see what you mean, thanks for clarifying!
I think this is doable, lets try it in https://github.com/conan-io/conan/pull/9843

**Comment by taogeng** [comment]
> Now I see what you mean, thanks for clarifying! I think this is doable, lets try it in #9843

Thanks very much.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/9843 merged, this will be in next 1.42
